pkg/network/ebpf/c/prebuilt/tracer.c

```diff
@@ -329,13 +336,36 @@ static __always_inline bool check_family(struct sock* sk, u16 expected_family) {
     return family == expected_family;
 }
 
-static __always_inline int read_conn_tuple(conn_tuple_t* t, struct sock* skp, u64 pid_tgid, metadata_mask_t type) {
-    t->saddr_h = 0;
-    t->saddr_l = 0;
-    t->daddr_h = 0;
-    t->daddr_l = 0;
-    t->sport = 0;
-    t->dport = 0;
+static __always_inline int get_proto(conn_tuple_t * t) {
+    return (t->metadata & CONN_TYPE_TCP) ? CONN_TYPE_TCP : CONN_TYPE_UDP;
+}
+
+static __always_inline void sockaddr_to_addr(struct sockaddr * sa, u64 * addr_h, u64 * addr_l, u16 * port) {
+    if (!sa) return;
+
+    u16 family;
+    bpf_probe_read(&family, sizeof(family), &sa->sa_family);
+
+    struct sockaddr_in * sin;
+    struct sockaddr_in6 * sin6;
+    switch (family) {
+    case AF_INET:
+        bpf_probe_read(&sin, sizeof(sin), &sa);
+        if (addr_l) bpf_probe_read(addr_l, sizeof(u32), &(sin->sin_addr.s_addr));
+        if (port) bpf_probe_read(port, sizeof(u16), &sin->sin_port);
+        break;
+    case AF_INET6:
+        bpf_probe_read(&sin6, sizeof(sin6), &sa);
+        if (addr_l && addr_h) {
+            bpf_probe_read(addr_h, sizeof(u64), &(sin6->sin6_addr.s6_addr));
```

Comment:
I think you either want `sin6->sin6_addr.s6_addr` or `&(sin6->sin6_addr.s6_addr[0])` right? `s6_addr` is already a pointer.

Reply:
Yes, correct. Good catch!